hedyorg/hedy
# [REFACTORING] use a json file for the babel caching

**Description**

Improve the developer experience of contributors by removing code generation which sometimes conflicts with linters / autoformatters.

We were generating python code for the i18n config in the frontend. Now we serialize to JSON and parse it instead, the python "API" stays the same.

The problems with generating python code were:
1) differences in the output when running `npm run build` (different formatting depending on the local system)
2) different ordering of the dicts depending on the local system
3) rewrites in the autopep8 pre-commit hook

With this commit we rewrite it to indented & ordered JSON so differences are simpler to review. Also the serialization is using 100% standard tooling instead of code generation.


**Fixes ....**

I ran into this when compiling frontend code and running `npm run build` while fixing https://github.com/hedyorg/hedy/issues/4475 .

I don't believe creating a separate issue for issues like this (which would create the exact same content as the PR) is needed, but if I'm wrong about this please let me know.

**How to test**

Check out the branch locally and check if multiple languages still display correctly

**Checklist**
